lib/ecto/changeset.ex

```diff
@@ -733,6 +743,17 @@ defmodule Ecto.Changeset do
       struct = model.__struct__
       query  = from m in struct, select: field(m, ^field), limit: 1
 
+      if opts[:scope] do
+        query = Enum.reduce(opts[:scope], query, fn(field, acc) ->
+          value = get_field(changeset, field)
+          if value do
```

Comment:
No, we don't ignore any. It is going to be much clearer like that, otherwise there will be bugs hiding. If you pass a field that does not exist, it should raise, and `field(m, ^field)` will raise automatically.